archive/100-theorems-list/83_friendship_graphs.lean

```diff
@@ -0,0 +1,311 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author:  Aaron Anderson, Jalex Stark, Kyle Miller.
+-/
+import combinatorics.adj_matrix
+import linear_algebra.char_poly.coeff
+import data.int.modeq
+import data.zmod.basic
+import tactic
+
+/-!
+# The Friendship Theorem
+
+## Definitions and Statement
+- A `friendship` graph is one in which any two distinct vertices have exactly one neighbor in common
+- A `politician`, at least in the context of this problem, is a vertex in a graph which is adjacent
+  to every other vertex.
+- The friendship theorem (Erdős, Rényi, Sós 1966) states that every finite friendship graph has a
+  politician.
```

Comment:
Could you give a full reference to the paper?

Is their proof the one you are formalising? If not, can you add references to what you are following?

Reply:
I've provided a link to the paper in question, and the paper with the proof that's closest to what we used. I'm going to have to figure out the standard format for full references.

Reply:
I think this is the right bibtex and all?